spacepy/irbempy/__init__.py

```diff
@@ -83,11 +83,2472 @@
+        '''Reassemble the results from the multiprocessing'''
+        funcs = {'Bmin': np.hstack,
+                 'Blocal': np.hstack,
+                 'Bmirr': np.vstack,
+                 'Lm': np.vstack,
+                 'Lstar': np.vstack,
+                 'MLT': np.hstack,
+                 'Xj': np.vstack}
+        out = dm.SpaceData()
+        for el in result:
+            for key in result[0]:
+                if key in list(out.keys()):  # not first chunk
+                    out[key] = funcs[key]([out[key], el[key]])
+                else:  # first chunk
+                    out[key] = el[key].copy()
+        return out
+
+    if isinstance(alpha, numbers.Number):
+        alpha = [alpha]
+
+    ncpus = spacepy.config['ncpus']
+    ncalc = len(ticks)
+    nalpha = len(alpha)
+
+    if ncalc < ncpus * 2:  # Don't multiprocess if not worth it
+        ncpus = 1
+    if ncpus > 1:
+        import __main__ as main
+        if hasattr(main, '__file__'):
+            try:
+                from multiprocessing import Pool
+                pool = Pool(ncpus)
+            except (ImportError, OSError):
+                ncpus = 1  # if pool setup fails, e.g. Wine, go single
+        else:
+            ncpus = 1  # won't multiprocess in interactive mode
+
+    if ncpus > 1:
+        nblocks = ncpus
+        blocklen = np.floor_divide(ncalc, ncpus)
+        tt, cc = [], []
+        if omnivals:
+            ov = []
+        else:
+            ov = [None] * nblocks
+        for block in range(nblocks):
+            startind = block*blocklen
+            if block != nblocks-1:  # not last block
+                endind = block*blocklen + blocklen
+            else:
+                endind = ncalc  # in last block, go to index of final time
+            tt.append(ticks[startind:endind])  # chunk time tags
+            cc.append(loci[startind:endind])  # chunk positions
+            if omnivals:
+                ov.append(get_ov(omnivals, startind, endind))
+        inputs = [[tch, cch, alpha, extMag, options, oval, landi2lstar]
+                  for tch, cch, oval in zip(tt, cc, ov)]
+        result = pool.map(_multi_get_Lstar, inputs)
+        pool.close()
+        pool.join()
+        DALL = reassemble(result)
+    else:  # single NCPU, no chunking
+        DALL = _get_Lstar(ticks, loci, alpha, extMag, options, omnivals, landi2lstar)
+
+    return DALL
+
+
+def _multi_get_Lstar(inputs):
+    '''
+    '''
+    ticks = inputs[0]
+    loci = inputs[1]
+    alpha = inputs[2]
+    extMag = inputs[3]
+    options = inputs[4]
+    omnivals = inputs[5]
+    landi2lstar = inputs[6]
+    DALL = _get_Lstar(ticks, loci, alpha, extMag, options, omnivals, landi2lstar)
+
+    return DALL
+
+
+# -----------------------------------------------
+def prep_irbem(ticks=None, loci=None, alpha=[], extMag='T01STORM', options=[1, 0, 0, 0, 0], omnivals=None):
+    """
+    Prepare inputs for direct IRBEM-LIB calls. Not expected to be called by the user.
+    """
```

Comment:
This needs to be documented properly

Reply:
Again old docs, out of scope for this transition.